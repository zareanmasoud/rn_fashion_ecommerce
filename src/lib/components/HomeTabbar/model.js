// @flow

export type Buttons = {
  press: Function,
  src: string,
  text: string,
};
