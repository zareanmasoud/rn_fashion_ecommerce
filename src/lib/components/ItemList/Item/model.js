// @flow

export type ItemModel = {
  id: string,
  photo: string,
};
