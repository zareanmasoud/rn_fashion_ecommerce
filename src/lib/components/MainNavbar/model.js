// @flow

export type Button = {
  action: Function,
  src: string,
};
