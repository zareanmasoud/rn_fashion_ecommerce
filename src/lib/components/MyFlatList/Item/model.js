// @flow

export type Data = {
  id: string,
  photo: string,
};
