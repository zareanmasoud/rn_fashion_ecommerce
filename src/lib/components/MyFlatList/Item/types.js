// @flow

export type Data = {
  id: string,
  photo: string,
};

export type Photo = string;
