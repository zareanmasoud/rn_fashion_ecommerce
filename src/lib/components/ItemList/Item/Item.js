import React from 'react';
import {Image, TouchableOpacity, View} from 'react-native';
// eslint-disable-next-line import/no-extraneous-dependencies
import {PRODUCT_SCREEN} from "screens/routes/constants";
import { withNavigation } from 'react-navigation';
import styles from "./styles";
import type {ItemModel} from "./model";

const renderItem = (item, navigation) => {
  return (
    <TouchableOpacity onPress={() => navigation.navigate(PRODUCT_SCREEN, {id: item.id})}>
      <View style={styles.itemView}>
        <Image source={item.photo} style={styles.itemImg}/>
      </View>
    </TouchableOpacity>
  )
};

type Props = {
  item: ItemModel,
};

const Item = ({item, navigation}: Props) => {
  return (
    <View style={styles.container}>
      {
        renderItem(item, navigation)
      }
    </View>
  )
};

export default withNavigation(Item);
