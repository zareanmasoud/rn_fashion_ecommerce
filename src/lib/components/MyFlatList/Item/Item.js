import React from 'react';
import {Image, TouchableOpacity, View} from 'react-native';
import type {Photo} from './types';
import styles from './styles';

type Props = {
  photo: Photo,
  handlePress: Function,
  index: number,
};

const Item = ({photo, handlePress, index}: Props) => {
  return (
    <TouchableOpacity testID={`${index}`} onPress={() => handlePress()}>
      <View style={styles.itemView}>
        <Image source={photo} style={styles.itemImg} />
      </View>
    </TouchableOpacity>
  );
};

export default Item;
