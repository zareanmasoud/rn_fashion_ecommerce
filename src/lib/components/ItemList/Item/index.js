import Item from './Item';

export {Item as default}
