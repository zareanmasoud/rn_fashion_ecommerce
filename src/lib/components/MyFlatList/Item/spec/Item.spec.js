import React from 'react';
import {shallow} from 'enzyme';
import Item from '../Item';
import model from './model';

const mockHandlePress = jest.fn();
mockHandlePress.mockReturnValue('Link on press invoked');
const enhancedModel = model.item(mockHandlePress);

describe('Item Component', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(<Item {...enhancedModel} index={0} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
