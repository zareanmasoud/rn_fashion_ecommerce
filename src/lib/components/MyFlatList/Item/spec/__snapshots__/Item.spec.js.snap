// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component Rendering should match to snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  testID="0"
>
  <View
    style={Object {}}
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 250,
          "width": 200,
        }
      }
    />
  </View>
</TouchableOpacity>
`;
